### ISSUE TITLE
Upgrade to v2.7.12+ issue

### ISSUE THREAD

**Issue by askripalev** [sijms/go-ora#435]
Hi!

After upgrade from v2.7.11 to v2.7.12 or later some queries stop working and error "ORA-01008: not all variables bound" occurs.

This won't work now:
```
	params := []any{
		sql.Named("test1", 44),
		sql.Named("test2", 2),
		sql.Named("test3", 300),
	}

	_, err := conn.Query("select :test1||:test1||:test2||:test3 from dual", params...)
	if err != nil {
		fmt.Println("DEBUG:", err.Error())
	}
```

Query like this is still working, of course:
```
...
	_, err := conn.Query("select :test1||:test2||:test3 from dual", params...)
...
```

And this works as well:

```
...
	_, err := conn.Query("select :test1||:test2||:test3||:test3 from dual", params...)
...
```

P.S. Code and query are much simplified, but still :)

**Comment by sijms** [comment]
named parameter is not implemented for query 
I wil fix in next release and inform you

**Comment by sijms** [comment]
I test you code with c# driver and result is same error
`ORA-01008: not all variables bound`

**Comment by askripalev** [comment]
The question is why the same code worked okay prior v2.7.12 (v2.7.11 and lower)? I'm using v2.7.9 now and have no described issue.

**Comment by sijms** [comment]
the way I use in v2.7.11 work with select but give error with pl/sql 
so to fix the issue I should merge the 2 solution 
sorry I didn't add the solution in v2.7.18 because it need more testing before release
but I will add it soon and publish new release

**Comment by sijms** [comment]
fixed in v2.7.22
